examples/module1/lesson1/validate-it/validation/validator.ts

```diff
@@ -0,0 +1,47 @@
+import { ValidationMethod } from './validation-methods';
+
+enum ValidationMessage {
+  SUCCESS_VALIDATION = 'Valid',
+  FAIL_VALIDATION = 'Invalid',
+}
+
+export function validator(validationMethods: ValidationMethod[]) {
+  const validationInput = document.getElementById(
+    'input'
+  ) as HTMLInputElement | null;
+  const validationButton = document.getElementById('button');
+  const clearButton = document.getElementById('button2');
+  const validationResult = document.getElementById('result');
+
+  if (
+    !validationButton ||
+    !clearButton ||
+    !validationInput ||
+    !validationResult
+  ) {
+    throw new Error('Some html element not found');
+  }
+
+  validationButton.addEventListener('click', () => {
+    if (!validationInput.value) {
+      validationResult.innerHTML = ValidationMessage.FAIL_VALIDATION;
+    }
+
+    const valueToValidate = Number(validationInput.value);
+
+    if (Number.isNaN(valueToValidate) && Number.isInteger(valueToValidate)) {
+      validationResult.innerHTML = ValidationMessage.FAIL_VALIDATION;
+    }
+
+    if (validationMethods.every((method) => method(valueToValidate))) {
```

Comment:
Spróbuj doprowadzić kod do takiej postaci:

```suggestion
    return validationMethods.every((method) => method(valueToValidate)
```